plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt

```diff
@@ -63,37 +85,79 @@ class SsoAccessTokenProvider(
             return it
         }
 
-        val token = pollForToken()
+        val token = if (isNewAuthPkce) {
+            ToolkitOAuthService.getInstance().authorize(registerPkceClient()).get()
+        } else {
+            pollForDAGToken()
+        }
 
         saveAccessToken(token)
 
         return token
     }
 
-    private fun registerClient(): ClientRegistration {
-        loadClientRegistration()?.let {
+    @Deprecated("Device authorization grant flow is deprecated")
+    private fun registerDAGClient(): ClientRegistration {
+        loadDagClientRegistration()?.let {
             return it
         }
 
         // Based on botocore: https://github.com/boto/botocore/blob/5dc8ee27415dc97cfff75b5bcfa66d410424e665/botocore/utils.py#L1753
         val registerResponse = client.registerClient {
-            it.clientType(CLIENT_REGISTRATION_TYPE)
+            it.clientType(PUBLIC_CLIENT_REGISTRATION_TYPE)
             it.scopes(scopes)
             it.clientName("AWS Toolkit for JetBrains")
         }
 
-        val registeredClient = ClientRegistration(
+        val registeredClient = DeviceAuthorizationClientRegistration(
+            registerResponse.clientId(),
+            registerResponse.clientSecret(),
+            Instant.ofEpochSecond(registerResponse.clientSecretExpiresAt()),
+            scopes
+        )
+
+        saveClientRegistration(registeredClient)
+
+        return registeredClient
+    }
+
+    private fun registerPkceClient(): PKCEClientRegistration {
+        loadPkceClientRegistration()?.let {
+            return it
+        }
+
+        if (!ssoUrl.contains("identitycenter")) {
```

Comment:
Should we have a test for this if we are doing text parsing?

Reply:
it's just a warning, service is right place to make the final call